clients/py/sapphirepy/sapphire.py

```diff
@@ -69,79 +93,187 @@ def _encrypt_tx_params(pk:CalldataPublicKey, params:tuple[TxParams]):
     else:
         raise TypeError("Invalid 'data' type", type(data))
     encrypted_data = c.encrypt(data_bytes)
-    params[0]['data'] = HexStr('0x' + hexlify(encrypted_data).decode('ascii'))
+
+    if params[0]['from'] and method == 'eth_call':
+        assert account is not None, 'Account must be provided for encrypted calls!'
```

Comment:
If no `account` is provided we can do regular `eth_call` with non-signed but encrypted calldata.

If `account` is provided, we can sign the call.

Reply:
Added account check to if statement, now signs data only when account is provided.